crates/proof-of-sql/src/base/database/column_index_operation.rs

```diff
@@ -0,0 +1,157 @@
+                )?;
+                Ok(Column::TinyInt(alloc.alloc_slice_copy(&raw_values) as &[_]))
+            }
+            ColumnType::SmallInt => {
+                let raw_values = Self::op(
+                    column.as_smallint().expect("Column types should match"),
+                    indexes,
+                )?;
+                Ok(Column::SmallInt(alloc.alloc_slice_copy(&raw_values) as &[_]))
+            }
+            ColumnType::Int => {
+                let raw_values =
+                    Self::op(column.as_int().expect("Column types should match"), indexes)?;
+                Ok(Column::Int(alloc.alloc_slice_copy(&raw_values) as &[_]))
+            }
+            ColumnType::BigInt => {
+                let raw_values = Self::op(
+                    column.as_bigint().expect("Column types should match"),
+                    indexes,
+                )?;
+                Ok(Column::BigInt(alloc.alloc_slice_copy(&raw_values) as &[_]))
+            }
+            ColumnType::Int128 => {
+                let raw_values = Self::op(
+                    column.as_int128().expect("Column types should match"),
+                    indexes,
+                )?;
+                Ok(Column::Int128(alloc.alloc_slice_copy(&raw_values) as &[_]))
+            }
+            ColumnType::Scalar => {
+                let raw_values = Self::op(
+                    column.as_scalar().expect("Column types should match"),
+                    indexes,
+                )?;
+                Ok(Column::Scalar(alloc.alloc_slice_copy(&raw_values) as &[_]))
+            }
+            ColumnType::Decimal75(precision, scale) => {
+                let raw_values = Self::op(
+                    column.as_decimal75().expect("Column types should match"),
+                    indexes,
+                )?;
+                Ok(Column::Decimal75(
+                    precision,
+                    scale,
+                    alloc.alloc_slice_copy(&raw_values) as &[_],
+                ))
+            }
+            ColumnType::VarChar => {
+                let (raw_values, raw_scalars) =
+                    column.as_varchar().expect("Column types should match");
+                let raw_values = Self::op(raw_values, indexes)?;
+                let scalars = Self::op(raw_scalars, indexes)?;
+                Ok(Column::VarChar((
+                    alloc.alloc_slice_clone(&raw_values) as &[_],
+                    alloc.alloc_slice_copy(&scalars) as &[_],
+                )))
+            }
+            ColumnType::TimestampTZ(tu, tz) => {
+                let raw_values = Self::op(
+                    column.as_timestamptz().expect("Column types should match"),
+                    indexes,
+                )?;
+                Ok(Column::TimestampTZ(
+                    tu,
+                    tz,
+                    alloc.alloc_slice_copy(&raw_values) as &[_],
+                ))
+            }
+        }
+    }
+}
+
+pub struct ApplyIndexOp {}
```

Comment:
Why not use a unit struct?
```suggestion
pub struct ApplyIndexOp;
```